tools/stonith_admin.c

```diff
@@ -639,30 +633,40 @@ main(int argc, char **argv)
     switch (action) {
         case 'I':
             rc = st->cmds->list_agents(st, st_opt_sync_call, NULL, &devices, timeout);
+
+            out->begin_list(out, "Installed fence devices", "fence device", "fence devices");
             for (dIter = devices; dIter; dIter = dIter->next) {
-                fprintf(stdout, " %s\n", dIter->value);
+                out->list_item(out, "device", dIter->value);
             }
+
             if (rc == 0) {
-                fprintf(stderr, "No devices found\n");
+                out->end_list(out);
             } else if (rc > 0) {
-                fprintf(stderr, "%d devices found\n", rc);
+                out->end_list(out);
                 rc = 0;
```

Comment:
You could simplify it as 
```
rc = 0;
out->end_list(out);
```

Reply:
IOW the new model means we don't need an if-else here or similarly below